src/psyclone/tests/domain/lfric/lfric_scalar_test.py

```diff
@@ -0,0 +1,127 @@
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: I. Kavcic, Met Office
+
+'''
+Module containing pytest tests for PSy-layer code generation and the related
+functionality for the LFRic scalar arguments.
+'''
+
+# Imports
+from __future__ import absolute_import, print_function
+import os
+import pytest
+from psyclone.domain.lfric import LFRicArgDescriptor
+from psyclone.dynamo0p3 import LFRicScalarArgs
+from psyclone.f2pygen import ModuleGen
+from psyclone.parse.algorithm import parse
+from psyclone.psyGen import PSyFactory
+from psyclone.errors import InternalError, GenerationError
+
+# Constants
+BASE_PATH = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.abspath(__file__)))),
+    "test_files", "dynamo0p3")
+TEST_API = "dynamo0.3"
+
+# General scalar checks (argument type, data type, etc)
+
+
+def test_lfricscalars_call_err():
+    ''' Check that the LFRicScalarArgs constructor raises the expected
+    internal error if it encounters an unrecognised intrinsic type of
+    scalar when generating a kernel call.
+
+    '''
+    _, invoke_info = parse(
+        os.path.join(BASE_PATH,
+                     "1.7_single_invoke_2scalar.f90"),
+        api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=True).create(invoke_info)
+    invoke = psy.invokes.invoke_list[0]
+    kernel = invoke.schedule.coded_kernels()[0]
+    # Sabotage the scalar argument to make it have an invalid intrinsic type
+    scalar_arg = kernel.arguments.args[0]
+    scalar_arg._intrinsic_type = "double-type"
+    with pytest.raises(InternalError) as err:
+        LFRicScalarArgs(invoke)._invoke_declarations(ModuleGen(name="my_mod"))
+    assert ("Found unsupported intrinsic types in Invoke "
+            "'invoke_0_testkern_two_scalars_type' declarations for the "
+            "scalar arguments ['a']. Supported types are "
+            "['real', 'integer']." in str(err.value))
+
+
+def test_int_real_scalar_invalid():
+    ''' Tests that the same scalar cannot have different data types
+    in different kernels within the same Invoke.
+
+    '''
+    _, invoke_info = parse(
+        os.path.join(BASE_PATH,
+                     "4.16_multikernel_invokes_real_int_scalar_invalid.f90"),
+        api=TEST_API)
+    psy = PSyFactory(TEST_API, distributed_memory=False).create(invoke_info)
+
+    with pytest.raises(GenerationError) as err:
+        _ = psy.gen
+    assert ("At least one scalar (['b']) in Invoke "
```

Comment:
s/At least one scalar/Scalar argument(s) ['b'] in Invoke.../

Reply:
Changed messages for scalar and field arguments.